### DISCUSSION TITLE
How to set s3 files to public?

### DISCUSSION THREAD

**Discussion by jelmore1674** [keystonejs/keystone#7853] [Questions]
I'm trying to find a way to when I upload an image with s3. Is there a way to just set it to public where I don't have to pass a signature with expiry? Like I was going through the docs and on v5, you could use the `uploadParams` to set `acl:'read-right'`. But for v6 I cannot find a solution. Maybe I'm missing something, but I would like help. Thanks in advance!

**Comment by elliott-w** [comment]
Yeah it's a doozy, for my purposes I've had to create 4 different lists: 

- PrivateFile and PublicFile (using file field)
- PrivateImage and PublicImage (using image field)

And each of them uses a separate storage string as defined in `keystone.ts` config:
```ts
config({
  // Other config here
  storage: {
    s3_images_private: {
      kind: 's3',
      type: 'image',
      bucketName: S3_BUCKET_NAME,
      region: S3_REGION,
      accessKeyId: S3_ACCESS_KEY_ID,
      secretAccessKey: S3_SECRET_ACCESS_KEY,
      signed: {
        expiry: 3600,
      },
      pathPrefix: 'private/images/',
    },
    s3_images_public: {
      kind: 's3',
      type: 'image',
      bucketName: S3_BUCKET_NAME,
      region: S3_REGION,
      accessKeyId: S3_ACCESS_KEY_ID,
      secretAccessKey: S3_SECRET_ACCESS_KEY,
      pathPrefix: 'public/images/',
    },
    s3_files_private: {
      kind: 's3',
      type: 'file',
      bucketName: S3_BUCKET_NAME,
      region: S3_REGION,
      accessKeyId: S3_ACCESS_KEY_ID,
      secretAccessKey: S3_SECRET_ACCESS_KEY,
      signed: {
        expiry: 3600,
      },
      pathPrefix: 'private/files/',
    },
    s3_files_public: {
      kind: 's3',
      type: 'file',
      bucketName: S3_BUCKET_NAME,
      region: S3_REGION,
      accessKeyId: S3_ACCESS_KEY_ID,
      secretAccessKey: S3_SECRET_ACCESS_KEY,
      pathPrefix: 'public/files/',
    },
  }
})
```

So just don't specifiy the `signed` property on your storage objects and keystone won't try to sign them (and add all those query variables to the url). You'll also need to make sure you have an s3 bucket policy which makes files in the `/public` folder publicly accessible. Here's mine:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Statement1",
            "Effect": "Allow",
            "Principal": "*",
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3:::bucket-name-here/public/*"
        }
    ]
}
```

Hope that helps, took me a while to figure it all out!
